packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.tsx

```diff
@@ -289,6 +300,7 @@ const DateRangeCalendar = React.forwardRef(function DateRangeCalendar<TDate>(
   const maxDateWithDisabled = (disabled && value[1]) || maxDate;
 
   const [rangePreviewDay, setRangePreviewDay] = React.useState<TDate | null>(null);
+  const [rangeDragDay, setRangeDragDay] = React.useState<TDate | null>(null);
```

Comment:
Maybe I'm missing something, but personally I would move this start inside `useDragRange` and return the value from the hook instead of passing the setter to the hook.

That way we encapsulate all of the logic

Reply:
I've structured it like this to have the least amount of frequently changing variables being shared among component and hook in order to facilitate least amount of hook result re-creations.
But I've experimented and I think splitting the hook into a sub-hook only for event handling helps in this case. 🤔

Reply:
Updated: https://github.com/mui/mui-x/pull/6763/commits/d388bcd293d7d32f82605ea4c0d7c71892d05e4e